tools/Google.Cloud.Tools.DocUploader/Program.cs

```diff
@@ -0,0 +1,79 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at 
+//
+// https://www.apache.org/licenses/LICENSE-2.0 
+//
+// Unless required by applicable law or agreed to in writing, software 
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and 
+// limitations under the License.
+
+using CommandLine;
+using Google.Cloud.Tools.DocUploader;
+using Google.Protobuf;
+using Google.Protobuf.WellKnownTypes;
+
+
+Parser.Default.ParseArguments<CreateMetadataOptions, UploadFileOptions>(args)
+    .WithParsed<CreateMetadataOptions>(options => CreateMetadata(options))
+    .WithParsed<UploadFileOptions>(options => UploadFile(options));
+
+// Creates the Metadata object based on the specified options and then store it as a JSON file at options.Destination.
+void CreateMetadata(CreateMetadataOptions options)
+{
+    var metadata = new Metadata();
+    metadata.UpdateTime = Timestamp.FromDateTimeOffset(DateTimeOffset.UtcNow);
+    metadata.Name = options.PackageName;
+    metadata.Version = options.version;
+    metadata.Language = options.Language;
+    metadata.DistributionName = options.DistributionName;
+    metadata.ProductPage = options.ProductPage;
+    metadata.GithubRepository = options.GitHubRepo;
+    metadata.IssueTracker = options.IssueTracker;
+    metadata.Stem = options.Stem;
+    metadata.Xrefs.Add(options.Xrefs);
+    metadata.XrefServices.Add(options.XrefServices);
+
+    File.WriteAllText(options.Destination, metadata.ToString());
+}
+
+// Uploads the zip file to the specified storage bucket.
+void UploadFile(UploadFileOptions options)
+{
+    string metadataFilePath = options.MetadataFile;
+    if (String.IsNullOrEmpty(metadataFilePath))
+    {
+        var jsonFile = Path.Combine(options.DocumentationPath, "docs.metadata.json");
+        if (File.Exists(jsonFile))
+        {
+            metadataFilePath = jsonFile;
+        }
+        else
+        {
+            throw new Exception("You need metadata to upload the docs.You can generate it with docuploader create-metadata");
+        }
+    }
+    if (Directory.EnumerateFiles(options.DocumentationPath).ToList().Count == 0)
+    {
+        throw new Exception($"The documentation path given({options.DocumentationPath}) does not contain any documentation files.");
```

Comment:
May be we can re-write it as $"The documentation path `{options.DocumentationPath}` is empty. Please provide a path to a folder that contains documentation files."